include/net/ilnp6.h

```diff
@@ -147,8 +147,8 @@ struct ilcc_table {
 /*Structure for LU message*/
 struct lu_data {
       uint64_t		l64;
-      uint32_t		prec;
-      uint32_t		lifetime;
+      uint16_t		prec;
+      uint16_t		lifetime;
```

Comment:
I think we need to consider the following observation. 
According to RFC 6743, the lifetime is the maximum number of seconds that this particular locator value is considered valid, this is identical to the DNS lifetime of the locator. 
I think, here, the DNS lifetime is considered to be the TTL.
According to the DNS RFC 1034, the TTL is a 32-bit number. 
@saleembhatti please advise.

Reply:
Understood. I'm just drawing attention to the fact that the 'reference' implementation does not match the RFC...